lib/vmpooler/pool_manager.rb

```diff
@@ -265,8 +269,8 @@ def clone_vm(pool_name, provider)
 
     def _clone_vm(pool_name, provider)
       # Generate a randomized hostname
-      o = [('a'..'z'), ('0'..'9')].map(&:to_a).flatten
-      new_vmname = $config[:config]['prefix'] + o[rand(25)] + (0...14).map { o[rand(o.length)] }.join
+      random_name = [@name_generator.adjective(max: 7), @name_generator.noun(max: 7)].join('-')
+      new_vmname = $config[:config]['prefix'] + random_name
```

Comment:
@highb This line would allow the name to exceed 15 characters... how about we modify this like so:

```ruby
netbios_limit = 15
adj_len = (netbios_limit - $config[:config]['prefix'].length) / 2
random_name = [@name_generator.adjective(max: adj_len), @name_generator.noun(max: adj_len)].join('-')
new_vmname = $config[:config]['prefix'] + random_name
```

Reply:
This doesn't seem to be a problem with the current 15 character random name. What does NetBIOS do if the name exceeds 15 characters? Truncate it? Do we actually use prefixes on any pools?

Reply:
I asked/looked around and we don't use the prefix. The prefix config value was added by a community member so I assume they don't mind if the hostname exceeds 15 characters.
Here's the relevant commit: https://github.com/puppetlabs/vmpooler/commit/10e507c262ae496c88b31e5c03d126ecc51addf0

Reply:
We have customers running this who I know for a fact use Windows machines so I thought it best to make sure that if we were aiming for a 15 character or less name that that was enforced even if those users add a prefix.

Reply:
The problem with the `(netbios_limit - $config[:config]['prefix'].length) / 2` formula is that _I think_ it is going to alternate between being 15 and 16 characters depending on if the prefix is odd or even due to integer rounding and the hypen. See this little pry experiment:
```
[10] pry(main)> prefix_len = 0; (((15 - prefix_len) / 2) * 2) + 1 + prefix_len
=> 15
[11] pry(main)> prefix_len = 1; (((15 - prefix_len) / 2) * 2) + 1 + prefix_len
=> 16
[12] pry(main)> prefix_len = 2; (((15 - prefix_len) / 2) * 2) + 1 + prefix_len
=> 15
[13] pry(main)> prefix_len = 3; (((15 - prefix_len) / 2) * 2) + 1 + prefix_len
=> 16
[14] pry(main)> prefix_len = 4; (((15 - prefix_len) / 2) * 2) + 1 + prefix_len
=> 15
[15] pry(main)> prefix_len = 5; (((15 - prefix_len) / 2) * 2) + 1 + prefix_len
=> 16
```
So we'd need a different formula there. I'm not sure if we should put in the effort of adding fiddly logic like this if users seem to be fine with the current prefix plus 15 character length.

Reply:
It looks like changing to `adj_len = ((netbios_limit - prefix.length) / 2) - 1` fixes that @highb. I tested with this:

```ruby
def new_name_info (prefix)
  netbios_limit = 15
  adj_len = ((netbios_limit - prefix.length) / 2) - 1
  # random_name = [@name_generator.adjective(max: adj_len), @name_generator.noun(max: adj_len)].join('-')
  random_name = ['a' * adj_len, 'a' * adj_len].join('-')
  new_vmname = prefix + random_name
  puts "With a #{prefix.length}-charcater prefix, the new vmname is #{new_vmname} and is #{new_vmname.length} characters long"
end

new_name_info('')
new_name_info('b')
new_name_info('bb')
new_name_info('bbb')
new_name_info('bbbb')
new_name_info('bbbbb')
```

```
╔ ☕️  gene:~
╚ᐅ ruby ~/Downloads/test.rb
With a 0-charcater prefix, the new vmname is aaaaaa-aaaaaa and is 13 characters long
With a 1-charcater prefix, the new vmname is baaaaaa-aaaaaa and is 14 characters long
With a 2-charcater prefix, the new vmname is bbaaaaa-aaaaa and is 13 characters long
With a 3-charcater prefix, the new vmname is bbbaaaaa-aaaaa and is 14 characters long
With a 4-charcater prefix, the new vmname is bbbbaaaa-aaaa and is 13 characters long
With a 5-charcater prefix, the new vmname is bbbbbaaaa-aaaa and is 14 characters long
```

Reply:
That has the same problem, and reduces the corpus of random words by a large factor. I really don't think this is worth worrying about until someone that actually uses the prefix config setting opens an issue explaining their use case.

Reply:
Feels like a spec would work well here

One edge case is if the prefix is large I'm not sure what the generator does with maxes set to 0, 1 or 2

While we are at it we could also adjust the noun's max length after we know the size of the adjective eg if the adjective is 3 chars then the the noun can be 11 char max?